wfe2/wfe.go

```diff
@@ -1617,8 +1616,18 @@ func (wfe *WebFrontEndImpl) Certificate(ctx context.Context, logEvent *web.Reque
 			return
 		}
 
-		// TODO(#5225): Check that the signature on parsedCert validates from the
-		// issuer cert in the chain.
+		// Double check that the signature validates.
```

Comment:
Yeah, this is definitely just belt+suspenders work. The potential mixup is something like "what if a bug causes us to construct one of those maps with the wrong IssuerNameID (maybe the ID of the root instead of the lowest intermediate)". Those maps are constructed from a config at startup; this check happens for every request in real time. It's nice to have the redundancy.